source/base/parameter_handler.cc

```diff
@@ -1140,10 +1161,16 @@ ParameterHandler::set(const std::string &entry_string, const bool new_value)
 std::ostream &
 ParameterHandler::print_parameters(std::ostream &    out,
                                    const OutputStyle style,
-                                   const bool        sort_alphabetical) const
+                                   const bool        sort_alphabetical,
+                                   const bool        print_documentation) const
 {
   AssertThrow(out, ExcIO());
 
+  Assert((style == Text || style == LaTeX || style == Description ||
+          style == ShortText) ^
+           !print_documentation,
```

Comment:
Ugh, xor and negation! Nobody will be able to read this :-) Just write it as `(A && !B) || (!A && B)`.